webservices/legal_docs/archived_murs.py

```diff
@@ -0,0 +1,386 @@
+                        us_code_title, us_code_section, us_code_match.group("paragraphs"))
+                    url = "https://www.govinfo.gov/link/uscode/{0}/{1}".format(us_code_title, us_code_section)
+
+                    us_codes.append({"text": citation_text, "url": url})
+
+                elif regulation_match:
+                    url = utils.create_eregs_link(regulation_match.group("part"), regulation_match.group("section"))
+                    regulations.append({"text": row["cite"], "url": url})
+                else:
+                    raise Exception("Could not parse archived mur's citation.")
+        return {"us_code": us_codes, "regulations": regulations}
+
+
+def get_subjects(mur_id):
+    subject_lv1 = []
+    subject_lv2 = []
+    subject_lv3 = []
+    with db.engine.connect() as conn:
+        rs1 = conn.execute(MUR_SUBJECT_LV1, mur_id)
+        for row1 in rs1:
+            if row1["subject"]:
+                rs2 = conn.execute(MUR_SUBJECT_LV2, mur_id, row1["level_1"])
+                subject_lv2 = []
+                for row2 in rs2:
+                    if row2["subject"]:
+                        rs3 = conn.execute(MUR_SUBJECT_LV3, mur_id, row1["level_1"], row2["level_2"])
+                        subject_lv3 = []
+                        for row3 in rs3:
+                            if row3["subject"]:
+                                subject_lv3.append({"text": row3["subject"]})
+
+                        if subject_lv3:
+                            subject_lv2.append({"text": row2["subject"], "children": subject_lv3})
+                        else:
+                            subject_lv2.append({"text": row2["subject"]})
+                if subject_lv2:
+                    subject_lv1.append({"text": row1["subject"], "children": subject_lv2})
+                else:
+                    subject_lv1.append({"text": row1["subject"]})
+    return subject_lv1
+
+
+def get_documents(mur_id):
+    documents = []
+    with db.engine.connect() as conn:
+        rs = conn.execute(MUR_DOCUMENTS, mur_id)
+        for row in rs:
+            documents.append({
+                "document_id": int(row["document_id"] or 1),
+                "length": int(row["length"] or 0),
+                "text": row["pdf_text"],
+                "url": (row["url"] or "/files/legal/murs/{0}.pdf".format(mur_id))
+            })
+    return documents
+
+
+def extract_pdf_text(mur_no=None):
+    """
+    1)Reads "text" and "documents" object data for Archived MURs from Elasticsearch,
+    under index: `archived_murs` and doc_type of `murs`
+    2)Assembles a JSON document corresponding to the archived murs,
+    3)Insert the JSON document into Postgres database table: mur_arch.documents
+    4)Run this command carefully, backup mur_arch.documents table first
+    and empty it. the data will be inserted into table: mur_arch.documents
+    """
+    es = utils.get_elasticsearch_connection()
+    each_fatch_size = 1000
```

Comment:
If we decide based on this comment https://github.com/fecgov/openFEC/pull/4577/files#r479442306 to go back in to the code, should this be `fetch` and not `fatch`?

Reply:
when the loop goes the last circle: start from no=4000, fatch size=1000, so should fatch 4000-5000 row.